### PR TITLE
Draft: Fit cubic port to JS

### PR DESCRIPTION
**Pull request by fatih-erikli**

### CODE REVIEW

**Comment by fatih-erikli** [comment]
Shall we add `zip` function to the utils? @justvanrossum It's used in some places in python version of the implementation.

File: src/fontra/client/core/vector.js [hunk 1]
```diff
@@ -7,7 +7,11 @@ export function subVectors(vectorA, vectorB) {
 }
 
 export function mulVector(vector, scalar) {
```
Comment (justvanrossum) [inline src/fontra/client/core/vector.js:9]: Instead of modifying this function, let's add a new one vor vector * vector called `mulVectorVector()`.

Comment (justvanrossum) [reply src/fontra/client/core/vector.js:9]: Additionally, we could consider renaming the original function to `mulVectorScalar()`.

File: src/fontra/client/core/vector.js [hunk 2]
```diff
@@ -61,3 +65,7 @@ export function intersect(pt1, pt2, pt3, pt4) {
 export function distance(pt1, pt2) {
   return vectorLength(subVectors(pt2, pt1));
 }
+
+export function dotVector(point, point2) {
```
Comment (justvanrossum) [inline src/fontra/client/core/vector.js:69]: Let's rename this `dotProduct()`, and the arguments `vectorA` and `vectorB` to keep the style consistend with the rest of the module.

**Review by justvanrossum — COMMENTED** [review]
First round of review.

File: src/fontra/client/core/vector.js (hunk 1 above)
Comment (fatih-erikli) [reply src/fontra/client/core/vector.js:9]: Ok. That makes sense

File: src/fontra/client/core/fit-cubic.js [hunk 3]
```diff
@@ -0,0 +1,151 @@
+import { enumerate, range } from "./utils.js";
+import {
+  subVectors,
```
Comment (justvanrossum) [inline src/fontra/client/core/fit-cubic.js:3]: Please list imported name in sorted order.

**Pull request merged**